crates/cdk-mintd/src/main.rs

```diff
@@ -318,6 +323,82 @@ async fn main() -> anyhow::Result<()> {
 
     mint_builder = mint_builder.add_cache(Some(cache.ttl.as_secs()), cached_endpoints);
 
+    // Add auth to mint
+    if let Some(auth_settings) = settings.auth {
+        let mint_blind_auth_endpoint =
+            ProtectedEndpoint::new(cdk::nuts::Method::Post, RoutePath::MintBlindAuth);
+
+        mint_builder = mint_builder.set_clear_auth_settings(
+            auth_settings.openid_discovery,
+            auth_settings.openid_client_id,
+            vec![mint_blind_auth_endpoint],
+        );
+
+        let mut protected_endpoints = HashMap::new();
+
+        let mut blind_auth_endpoints = vec![];
+
+        if auth_settings.enabled_mint {
+            let mint_quote_protected_endpoint = ProtectedEndpoint::new(
+                cdk::nuts::Method::Post,
+                cdk::nuts::RoutePath::MintQuoteBolt11,
+            );
+            protected_endpoints.insert(mint_quote_protected_endpoint, AuthRequired::Blind);
+            let mint_protected_endpoint =
+                ProtectedEndpoint::new(cdk::nuts::Method::Post, cdk::nuts::RoutePath::MintBolt11);
+
+            protected_endpoints.insert(mint_protected_endpoint, AuthRequired::Blind);
+
+            blind_auth_endpoints.push(mint_quote_protected_endpoint);
+            blind_auth_endpoints.push(mint_protected_endpoint);
+        }
+
+        if auth_settings.enabled_melt {
+            let melt_quote_protected_endpoint = ProtectedEndpoint::new(
+                cdk::nuts::Method::Post,
+                cdk::nuts::RoutePath::MeltQuoteBolt11,
+            );
+
+            protected_endpoints.insert(melt_quote_protected_endpoint, AuthRequired::Blind);
+
+            let melt_protected_endpoint =
+                ProtectedEndpoint::new(cdk::nuts::Method::Post, cdk::nuts::RoutePath::MeltBolt11);
+            protected_endpoints.insert(melt_protected_endpoint, AuthRequired::Blind);
+
+            blind_auth_endpoints.push(melt_quote_protected_endpoint);
+            blind_auth_endpoints.push(melt_protected_endpoint);
+        }
+
+        if auth_settings.enabled_swap {
+            let swap_protected_endpoint =
+                ProtectedEndpoint::new(cdk::nuts::Method::Post, cdk::nuts::RoutePath::Swap);
+            protected_endpoints.insert(swap_protected_endpoint, AuthRequired::Blind);
+
+            blind_auth_endpoints.push(swap_protected_endpoint);
+        }
+
+        mint_builder =
+            mint_builder.set_blind_auth_settings(auth_settings.mint_max_bat, blind_auth_endpoints);
+
+        let auth_localstore: Arc<dyn MintAuthDatabase<Err = cdk_database::Error> + Send + Sync> =
+            match settings.database.engine {
+                DatabaseEngine::Sqlite => {
+                    let sql_db_path = work_dir.join("cdk-mintd-auth.sqlite");
```

Comment:
Does this need to be a separate DB?

It makes DB backup and restore more complicated (2 files to backup or restore, on restore the correct pair of files has to be used).

Reply:
It makes sense to have two separate databases since, while there is large overlap between them, they serve different purposes. For example, the auth database may grow faster as keysets are likely to be rotated more frequently. This separation also helps with separation of concerns between the auth stuff and ecash stuff.

I don't think an extra file adds too much complexity. The auth database is also less critical to back up since if it's lost, the mint can simply create new auth keysets, require everyone to reauth, and generate new tokens.